src/app/pages/swap/hooks/use-fiat-price.tsx

```diff
@@ -0,0 +1,36 @@
+import { Money, createMoney } from '@shared/models/money.model';
+import { isUndefined } from '@shared/utils';
+
+import { useConvertAlexSwapCurrencyToFiatAmount } from '@app/common/hooks/use-convert-to-fiat-amount';
+import { i18nFormatCurrency } from '@app/common/money/format-money';
+import { unitToFractionalUnit } from '@app/common/money/unit-conversion';
+
+export function useAmountAsFiat(balance?: Money, price?: Money, value?: string) {
```

Comment:
This is named as though it's a generic conversion utility, but it isn't because it's specific to alex swaps.

Also, why is every prop conditional? If it's because the data is loaded async and isn't already ready, find a way to structure components such that by the time you need to use this hook you know that it is loaded. Would a loader component work? Perhaps use the router to ensure data is present.

Right now one can consume this as `const x = useAmountAsFiat()` and that isn't a legitimate use of it.

Reply:
Ok, will change name. The asset has to be initialized as undefined for form validation, not sure how to handle that then?

Reply:
I am thinking this isn't specific to alex swaps, is it? It is using the alex-sdk for price data, but couldn't their price data be used for other swaps? Or, will Magic have its own price data? Maybe I just actually named the helper functions here incorrectly? 🤔

Reply:
`useAlexSdkAmountAsFiat` and `convertAlexSdkCurrencyToUsd`, etc.?